src/s2wasm.h

```diff
@@ -160,19 +160,36 @@ class S2WasmBuilder {
   }
 
   int32_t getInt() {
-    int32_t ret = 0;
+    const char* loc = s;
```

Comment:
The `dump()` function dumps from the *current* position of `s` onwards, so the previous error handling prints the file *just after* the number that was parsed. Setting `s` only on success prevents this.